### DISCUSSION TITLE
How to enable CORS of a specific API route?

### DISCUSSION THREAD

**Discussion by JohnCido** [solidjs/solid-start#961] [Q&A]
Now I've added all CORS-related `<HttpHeader />` to the `root.tsx` so that Solid Start could response OPTIONS with correct headers. Is there a way to have a more fine-grained control over it?

**Comment by IKoshelev** [comment]
From a quick experiment - `OPTIONS` request does not even reach middleware, so, you'll probably need an API gateway in front of your app :-(
